src/spikeinterface/preprocessing/remove_artifacts.py

```diff
@@ -263,12 +263,14 @@ def get_traces(self, start_frame, end_frame, channel_indices):
             traces = self.parent_recording_segment.get_traces(start_frame, end_frame, channel_indices)
         traces = traces.copy()
 
-        mask = (self.triggers >= start_frame) & (self.triggers < end_frame)
+        pad = self.pad
+        if pad is None:
+            mask = (self.triggers >= start_frame) & (self.triggers < end_frame)
+        else:
+            mask = (self.triggers >= start_frame - pad[1]) & (self.triggers < end_frame)
```

Comment:
Also at the end? I think it would be:
```suggestion
            mask = (self.triggers >= start_frame - pad[1]) & (self.triggers < end_frame + pad[0])
```